tests/test_predictors/test_svm_word_predictor.py

```diff
@@ -0,0 +1,320 @@
+        THRESHOLD = -1.5    --> P: 0.9716437941036409 R: 1.0
+        THRESHOLD = -1.4    --> P: 0.9755705281460552 R: 0.9999554446622705
+        THRESHOLD = -1.0    --> P: 0.9866772193641999 R: 0.9998217786490822
+        THRESHOLD = -0.5    --> P: 0.9955352184633155 R: 0.9984405631794689
+        THRESHOLD = 0.0     --> P: 0.9985657299090135 R: 0.9926483692746391
+        THRESHOLD = 1.0     --> P: 0.9997759019944723 R: 0.8944929602566387
+        """
+        THRESHOLD = -1.5
+        preds_pos = self.classifier.batch_predict(
+            words=self.pos_words, threshold=THRESHOLD
+        )
+        self.assertEqual(len(preds_pos), len(self.pos_words))
+        preds_pos_as_ints = [int(r.is_edit is False) for r in preds_pos]
+        tp = sum(preds_pos_as_ints)
+        fn = len(preds_pos_as_ints) - tp
+
+        preds_neg = self.classifier.batch_predict(
+            words=self.neg_words, threshold=THRESHOLD
+        )
+        self.assertEqual(len(preds_neg), len(self.neg_words))
+        preds_neg_as_ints = [int(r.is_edit is True) for r in preds_neg]
+        tn = sum(preds_neg_as_ints)
+        fp = len(preds_neg_as_ints) - tn
+
+        self.assertEqual(tp + fn + tn + fp, len(preds_pos) + len(preds_neg))
+
+        p = tp / (tp + fp)
+        r = tp / (tp + fn)
+
+        # uncomment for debugging
+        # print(f"P: {p} R: {r}")
+
+        self.assertGreaterEqual(p, 0.9)
+        self.assertGreaterEqual(r, 0.9)
+
+    def test_get_features(self):
+        (
+            all_features,
+            word_id_to_feature_ids,
+        ) = self.classifier._get_features(words=self.pos_words)
+        self.assertEqual(len(word_id_to_feature_ids), len(self.pos_words))
+        self.assertEqual(
+            all_features.shape[0],
+            sum([len(feature) for feature in word_id_to_feature_ids.values()]),
+        )
+        (
+            all_features,
+            word_id_to_feature_ids,
+        ) = self.classifier._get_features(words=self.neg_words)
+        self.assertEqual(len(word_id_to_feature_ids), len(self.neg_words))
+        self.assertEqual(
+            all_features.shape[0],
+            sum([len(feature) for feature in word_id_to_feature_ids.values()]),
+        )
+
+
+class TestSVMWordPredictor(unittest.TestCase):
+    @classmethod
+    def setUp(self):
+        self.fixture_path = os.path.join(os.path.dirname(__file__), "../fixtures/")
+        self.predictor = SVMWordPredictor.from_path(
+            tar_path=os.path.join(
+                self.fixture_path, "svm_word_predictor/svm_word_predictor.tar.gz"
+            )
+        )
+        with open(
+            os.path.join(
+                self.fixture_path,
+                "types/20fdafb68d0e69d193527a9a1cbe64e7e69a3798__pdfplumber_doc.json",
+            )
+        ) as f_in:
+            doc_dict = json.load(f_in)
+            self.doc = Document.from_json(doc_dict=doc_dict)
+
+    def test_predict_no_hyphen(self):
```

Comment:
@cmwilhelm added this test to handle failure case u found

Reply:
tyty